output/clickhouse.go

```diff
@@ -224,3 +225,48 @@ func (c *ClickHouse) initSchema() (err error) {
 	log.Infof("%s: Prepare sql=> %s", c.taskCfg.Name, c.prepareSQL)
 	return nil
 }
+
+func (c *ClickHouse) ChangeSchema(newKeys *sync.Map) (err error) {
+	var sqls []string
+	var onCluster string
+	if c.taskCfg.DynamicSchema.Cluster != "" {
+		onCluster = fmt.Sprintf("ON CLUSTER %s", c.taskCfg.DynamicSchema.Cluster)
+	}
+	newKeys.Range(func(key, value interface{}) bool {
+		strKey := key.(string)
+		strVal := value.(string)
+		switch strVal {
+		case "int":
+			strVal = "Nullable(Int64)"
+		case "float":
+			strVal = "Nullable(Float64)"
+		case "string":
+			strVal = "Nullable(String)"
+		default:
+			err = errors.Errorf("%s: BUG: unsupported column type %s", c.taskCfg.Name, strVal)
+			return false
+		}
+		sql := fmt.Sprintf("ALTER TABLE %s.%s %s ADD COLUMN %s %s", c.chCfg.DB, c.taskCfg.TableName, onCluster, strKey, strVal)
+		sqls = append(sqls, sql)
+		return true
+	})
+	if err != nil {
+		return
+	}
+	if c.taskCfg.DynamicSchema.DistTableName != "" {
+		sqls = append(sqls, fmt.Sprintf("DROP TABLE %s.%s %s", c.chCfg.DB, c.taskCfg.DynamicSchema.DistTableName, onCluster))
```

Comment:
I think it's hard to keep the consistency.

1. `On CLUSTER DDL` is background task.
2. DROP the distribute table is dangerous in production env because we may need query the data.

I don't think it's good to have a dynamic schema in clickhouse_sinker. It introduces lots of unstable issues, and it's pretty complicated.

I would rater use the latest DataTypeMap.

Reply:
As we talked offline, this shall be treated as experiment feature.